File: biometric/src/main/java/dev/skomlach/biometric/compat/utils/hardware/AbstractHardware.java
package dev.skomlach.biometric.compat.utils.hardware;

import dev.skomlach.biometric.compat.BiometricAuthRequest;

abstract class AbstractHardware implements HardwareInfo {
    private final BiometricAuthRequest authRequest;

    AbstractHardware(BiometricAuthRequest authRequest) {
        this.authRequest = authRequest;
    }

    public final BiometricAuthRequest getBiometricAuthRequest() {
        return authRequest;
    }

    @Override
    public abstract boolean isHardwareAvailable();

    @Override
    public abstract boolean isBiometricEnrolled();

    @Override
    public abstract boolean isLockedOut();
}

File: biometric/src/main/java/dev/skomlach/biometric/compat/utils/hardware/AbstractHardware.kt
package dev.skomlach.biometric.compat.utils.hardware

import dev.skomlach.biometric.compat.BiometricAuthRequest

abstract class AbstractHardware(val biometricAuthRequest: BiometricAuthRequest) : HardwareInfo {
    abstract override val isHardwareAvailable: Boolean
    abstract override val isBiometricEnrolled: Boolean
    abstract override val isLockedOut: Boolean
}